libs/datasets/data_source.py

```diff
@@ -60,14 +60,31 @@ def local(cls) -> "DataSource":
     @lru_cache(None)
     def beds(self) -> LatestValuesDataset:
         """Builds generic beds dataset"""
-        return LatestValuesDataset.build_from_data_source(self)
+        return self.latest_values()
 
     @lru_cache(None)
     def population(self) -> LatestValuesDataset:
         """Builds generic beds dataset"""
-        return LatestValuesDataset.build_from_data_source(self)
+        return self.latest_values()
 
     @lru_cache(None)
     def timeseries(self) -> TimeseriesDataset:
-        """Builds generic beds dataset"""
-        return TimeseriesDataset.build_from_data_source(self)
+        """Build TimeseriesDataset from this data source."""
+        if set(self.INDEX_FIELD_MAP.keys()) != set(TimeseriesDataset.INDEX_FIELDS):
+            raise ValueError("Index fields must match")
+
+        return TimeseriesDataset.from_source(
+            self, fill_missing_state=self.FILL_MISSING_STATE_LEVEL_DATA
+        )
+
+    @lru_cache(None)
```

Comment:
I guess it doesn't matter too much, but I'd lean towards dropping the lru_cache here. We tried to rely on it more when it was trickier to query certain sets of data, but now that's not as much of an issue

Reply:
just on latest_values or timeseries too? I'm guessing it makes a big difference for timeseries and I like the consistency.

Reply:
With both LRU removed run time increases from 300 -> 384 seconds.